lib/dotcom/utils/service_date_time.ex

```diff
@@ -0,0 +1,228 @@
+  """
+  @spec service_date() :: Date.t()
+  @spec service_date(DateTime.t()) :: Date.t()
+  def service_date(date_time \\ now()) do
+    if date_time.hour < @service_rollover_time.hour do
+      Timex.shift(date_time, hours: -@service_rollover_time.hour)
+      |> coerce_ambiguous_time()
+      |> Timex.to_date()
+    else
+      Timex.to_date(date_time)
+    end
+  end
+
+  @doc """
+  The most specific service range for the given date_time.
+  A date_time for today can be in the range for today and this week.
+  But, we just return :today since that is the most specific range.
+  """
+  @spec service_range(DateTime.t()) :: service_range()
+  def service_range(date_time) do
+    Enum.find(
+      [
+        &service_past?/1,
+        &service_today?/1,
+        &service_this_week?/1,
+        &service_next_week?/1,
+        &service_later?/1
+      ],
+      fn f ->
+        f.(date_time)
+      end
+    )
+    |> Kernel.inspect()
+    |> Kernel.then(fn module -> Regex.run(~r/_(\w+)\?/, module) end)
+    |> List.last()
+    |> String.to_atom()
+  end
+
+  @doc """
+  Get the beginning of the service day for the day after the given date_time.
+  """
+  @spec beginning_of_next_service_day() :: DateTime.t()
+  @spec beginning_of_next_service_day(DateTime.t()) :: DateTime.t()
+  def beginning_of_next_service_day(datetime \\ now()) do
+    datetime
+    |> end_of_service_day()
+    |> Timex.shift(microseconds: 1)
+    |> coerce_ambiguous_time()
+  end
+
+  @doc """
+  Get the beginning of the service day for the given date_time.
+  """
+  @spec beginning_of_service_day() :: DateTime.t()
+  @spec beginning_of_service_day(DateTime.t()) :: DateTime.t()
+  def beginning_of_service_day(date_time \\ now()) do
+    date_time
+    |> service_date()
+    |> Timex.to_datetime(@timezone)
+    |> coerce_ambiguous_time()
+    |> Map.put(:hour, @service_rollover_time.hour)
+  end
+
+  @doc """
+  Get the end of the service day for the given date_time.
+  """
+  @spec end_of_service_day() :: DateTime.t()
+  @spec end_of_service_day(DateTime.t()) :: DateTime.t()
+  def end_of_service_day(date_time \\ now()) do
+    date_time
+    |> service_date()
+    |> Timex.to_datetime(@timezone)
+    |> coerce_ambiguous_time()
+    |> Timex.shift(days: 1, hours: @service_rollover_time.hour, microseconds: -1)
+    |> coerce_ambiguous_time()
+    |> Map.put(:hour, @service_rollover_time.hour - 1)
+  end
+
+  @doc """
+  Get a service range for the day of the given date_time.
+  """
+  @spec service_range_day() :: Utils.DateTime.time_range()
+  @spec service_range_day(DateTime.t()) :: Utils.DateTime.time_range()
+  def service_range_day(date_time \\ now()) do
+    beginning_of_service_day = beginning_of_service_day(date_time)
+    end_of_service_day = end_of_service_day(date_time)
+
+    {beginning_of_service_day, end_of_service_day}
+  end
+
+  @doc """
+  Get a service range for the week of the given date_time.
```

Comment:
What is the week boundary? I can't find anything in the tests, docs, or code to tell me O_o.

There are some tests that look like they test the week boundaries, but it looks like they also use the same week boundaries in their test setup, so I'm still not sure how to tell O_o.

Reply:
Okay - I see a comment now that says that Monday is the service week boundary.

(Non-blocking) A doctest would be helpful here, as executable verification that the code actually does this.